README.md

```diff
@@ -209,27 +209,64 @@ Authorization in `root:redhat-appstudio` and `root:redhat-hacbs` workspaces in C
+
+Create the secrets with the following commands:
+
+```
+$ ./hack/setup-monitoring.sh oauth2-secret prometheus-oauth2-proxy $PROMETHEUS_GITHUB_CLIENT_ID $PROMETHEUS_GITHUB_CLIENT_SECRET $PROMETHEUS_GITHUB_COOKIE_SECRET
+
+$ ./hack/setup-monitoring.sh oauth2-secret grafana-oauth2-proxy $GRAFANA_GITHUB_CLIENT_ID $GRAFANA_GITHUB_CLIENT_SECRET $GRAFANA_GITHUB_COOKIE_SECRET
+```
+
+The `PROMETHEUS_GITHUB_CLIENT_ID`/`PROMETHEUS_GITHUB_CLIENT_SECRET` and `GRAFANA_GITHUB_CLIENT_ID`/`GRAFANA_GITHUB_CLIENT_SECRET` value pairs must match an existing "OAuth application" on GitHub - see [OAuth apps](https://github.com/organizations/redhat-appstudio-sre/settings/applications) in the [Red Hat Appstudio SRE organization](https://github.com/organizations/redhat-appstudio-sre). The `PROMETHEUS_GITHUB_COOKIE_SECRET` and `GRAFANA_GITHUB_COOKIE_SECRET` can be generated using the [following instructions](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/overview#generating-a-cookie-secret).
+
+
+#### Grafana Datasources
+
+Grafana datasources contain the connection settings to the Prometheus instances. These datasources are stored in secrets in the `appstudio-workload-monitoring` of the **Grafana cluster**.
+
+The Prometheus endpoints used by Grafana are protected by an OAuth proxy running as a sidecar container and which checks that the incoming requests contain a valid token. A token is valid if it belongs to a service account of the **Prometheus cluster** and which has the RBAC permission to "get namespaces". This can be obtained with the `cluster-monitoring-view` cluster role.
+
+```
+$ ./hack/setup-monitoring.sh grafana-datasource-secret $DATASOURCE_NAME $PROMETHEUS_URL $GRAFANA_TOKEN
+```
+
+`DATASOURCE_NAME` is the name of the secret itself, as well as the base name of the YAML file it contains. Grafana supports datasources from multiple secrets, so their names must be unique.
+
+`PROMETHEUS_URL` is obtained from the route created for Prometheus in the `openshift-monitoring` and `appstudio-workload-monitoring` namespaces:
 
-```yaml
-   apiVersion: v1
-   kind: Secret
-   metadata:
-     namespace: appstudio-workload-monitoring
-     name: prometheus-proxy-config
-     labels:
-       provider: appstudio-kcp-workload
-   data:
-     client-id: <github-oauth-client-id-for-prometheus>
-     client-secret: <github-oauth-secret-for-prometheus>
-     cookie-secret: <github-oauth-cookie-secret-for-prometheus>
-   type: Opaque
 ```
+$ PROMETHEUS_URL=`oc get route/prometheus-k8s -n openshift-monitoring -o json | jq -r '.status.ingress[0].host'`
+
+$ PROMETHEUS_URL=`oc get route/prometheus -n appstudio-workload-monitoring -o json | jq -r '.status.ingress[0].host'`
+```
+
+`GRAFANA_TOKEN` is obtained by requesting a token for the `grafana-oauth` service account in the **Prometheus cluster**:
+```
+$ GRAFANA_TOKEN=`oc create token grafana-oauth -n appstudio-workload-monitoring`
+```
```

Comment:
Do we need to ask users to enter the $PROMETHEUS_URL and $GRAFANA_TOKEN or can the script calculate them automatically?

Reply:
In a multi-cluster setup, each pair of `$PROMETHEUS_URL` and `$GRAFANA_TOKEN` values will be retrieved from the workload cluster were Prometheus is deployed, but used in the "control plane" cluster where the Grafana datasource secret is going to be created. So for now I believe it's simpler to ask the user to do it manually.

Reply:
OK that makes sense.  I would recommend restructuring this section in the order that people would run the commands, to make it easier to follow.

> #### Grafana Datasources
> 
> Grafana datasources contain the connection settings to the Prometheus instances. These datasources are stored in secrets in the `appstudio-workload-monitoring` namespace of the **Grafana cluster**.
> 
> The Prometheus endpoints used by Grafana are protected by an OAuth proxy running as a sidecar container and which checks that the incoming requests contain a valid token. A token is valid if it belongs to a service account of the **Prometheus cluster** and which has the RBAC permission to "get namespaces". This can be obtained with the `cluster-monitoring-view` cluster role.
>
> In a multi-cluster setup, each pair of $PROMETHEUS_URL and $GRAFANA_TOKEN values will be retrieved from the workload cluster were Prometheus is deployed, but used in the "control plane" cluster where the Grafana datasource secret is going to be created.  Run the following commands to obtain the tokens from each workload cluster where Prometheus is deployed:

(and then the commands to get those $PROMETHEUS_URL and $GRAFANA_TOKEN values)

> Using the values you just obtained from the **Prometheus cluster**, run the following command on the **Grafana cluster**:
> 

(and then the grafana-datasource-secret command and explanation about the arbitrary DATASOURCE_NAME that you recently added)

> Repeat this procedure for each workload cluster where Prometheus is deployed.

Reply:
I've updated that section in 01b893b, could you give it another look and let me know if it's ok?

Reply:
Per our conversation in Slack, please add `kustomize build components/monitoring/base | oc apply -f - `

Reply:
I see that in the latest commit, thanks.